Assets/UnityIO.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+using System;
+using System.IO;
+using System.Collections;
+
+public delegate void UnityIORequestCallback(object userData, Response response);
+
+public class Response {
+    public string error;
+    public byte[] data;
+
+	public Response(string e, byte[] d) {
+		error = e;
+		data = d;
+	}
+	public Response(string e, string d) {
+		error = e;
+		data = Convert.FromBase64String(d);
+	}
+}
+
+public class UnityIO 
+{
+	public IEnumerator UnityIORequest(string url, UnityIORequestCallback callback, object userData) {
+
+		bool networkRequest = false;
+		Response response;
+		if (url.StartsWith("http")) {
```

Comment:
Do you think we can leverage the usage of the `URL` class that .NET provides? https://msdn.microsoft.com/en-us/library/system.uri(v=vs.110).aspx.

We would have those built in.